contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -551,22 +551,11 @@ RETURNS INTEGER AS
 'babelfishpg_tsql', 'object_id'
 LANGUAGE C STABLE;
 
-CREATE OR REPLACE FUNCTION sys.parsename (
-	object_name VARCHAR
-	,object_piece INT
-	)
-RETURNS VARCHAR AS $$
-/***************************************************************
-EXTENSION PACK function PARSENAME(x)
-***************************************************************/
-SELECT CASE
-		WHEN char_length($1) < char_length(pg_catalog.replace($1, '.', '')) + 4
-			AND $2 BETWEEN 1
-				AND 4
-			THEN reverse(split_part(reverse($1), '.', $2))
-		ELSE NULL
-		END $$ immutable LANGUAGE 'sql';
-        
+CREATE OR REPLACE FUNCTION sys.parsename(object_name character varying, object_piece int)
```

Comment:
I feel that your input type should be sys.varchar, please double check agains the specification.

Reply:
Yes, will make the changes.